File: Practice 8/Гарифуллин/library-backend/src/publishers/entities/publisher.entity.ts
import { Book } from "src/books/entities/book.entity";
import { Column, Entity, OneToMany, PrimaryColumn } from "typeorm";

@Entity()
export class Publisher {
  @PrimaryColumn({unique: true})
  name: string

  @Column()
  address: string

  @OneToMany(type => Book, book => book.publisher)
  books: Book[]
}

File: Practice 8/Yanal Yahya/LibraryRestAPI/src/main/java/com/example/LibraryRestAPI/entity/PublisherEntity.java
package com.example.LibraryRestAPI.entity;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "publisher")
public class PublisherEntity {
    @Id
    private String name;
    private String address;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "publisher")
    private List<BookEntity> books;

    public PublisherEntity() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String adress) {
        this.address = adress;
    }
}
